src/main/savestates.c

```diff
@@ -839,6 +839,12 @@ static int savestates_load_m64p(struct device* dev, char *filepath)
             curr += sizeof(uint32_t);
 #endif
         }
+
+        if (version >= 0x00010700)
+        {
+            dev->sp.fifo[0] = GETDATA(curr, struct sp_dma);
```

Comment:
You cannot serialize/deserialize a struct in one shot, you must do it field by field, otherwise run into the problem a compiler-specific struct layout (adding padding bytes for instance).

Also, you must provide a fallback (for instance empty fifo) for savestates < 1.7.

Reply:
I will update to reference each member. I don't think setting a fallback for savestates < 1.7 would be needed. If the save state is < 1.7, there can't be one of these DMAs in the event queue, so the fifo values have no meaning/use